### ISSUE TITLE
Fix false positives for unsupported syntaxes in embedded style tags

### ISSUE THREAD

**Issue by Mister-Hope** [stylelint/stylelint#4719]
> Clearly describe the bug

Any vue files using stylus:

```xml
<style lang="stylus">
</stylus>
```

> What code is needed to reproduce this issue?

e.g.

```xml
<style lang="stylus">
.navbar
  position fixed
  z-index 20
  top 0
  left 0
  right 0
  height $navbarHeight
  padding $navbar-vertical-padding $navbar-horizontal-padding
  background-color var(--background-color)
  box-sizing border-box
  line-height $navbarHeight - 1.4rem
  box-shadow 0 2px 8px var(--box-shadow-color)
</stylus>
```

> What vscode-stylelint configuration is needed to reproduce the bug?

the default option, which contains

```
"vue",
"vue-html",
"vue-postcss",
```

> Is this issue related to autofix? (`editor.codeActionsOnSave`)

No

> Which version of vscode-stylelint are you using?

`0.84.0`

> Which version of stylelint are you using?

None

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes stylus

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

Unknown word (CssSyntaxError)

I supposed that stylelint should not try to lint the code it don't understand. Especially those it contains in its default list. At least it should check first. It seems that stylelint does not support stylus. Then we can just judge if it is stylus stynax and just ignore it when it is.

Throwing errors everywhere could be really annoying.

**Comment by jeddy3** [comment]
@Mister-Hope Thanks for the report and for using the template. I've transferred the issue to the main repo as its reproducible in stylelint itself [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIA8BnALgTwDYwARYEMA7AcwF4AdEdbAVxSoD4KiA6IggNwCMCAnFrlwAHCCgCWacRCK4AZuIAeMKINwAvALTiisRbgBMABjVoIw3Cdn4YctJbV9xJABb2rQlzGdvcAEg4efgAJb1c0NWECKCgdEn9A3j5NThg+KTACLE0omLiEriTNFwgndRk0LJzo2NI1XjAAaxI+CFpdTUgsUtxOfgAKTU0G5tb2qE6Ibr4ASnqIRU0JdXzuUthktcU1LB0YYrDfAMKQw-tNXABGVgAWPhgAW3nFlBdoiAB3S0NhfQAOX69AZDLZLN5QT6TaZzIhIAD0NCw9EYIAANNQMERKooECA3A8sGjwDIFCRccA1FQ+LQcAwELgKdYhGoAL4sFkgFlAA).

> I supposed that stylelint should not try to lint the code it don't understand.

I agree. It should only lint `<style>` tags:

- without a `lang` attribute, defaulting to `css` syntax
- with a `lang` attribute that matches [a supported syntax](https://stylelint.io/user-guide/usage/options#syntax)

I think the problem lies one of two upstream packages:

- postcss-syntax
- postcss-html

Neither has been updated in awhile and we may have to fork them into the stylelint organisation, as we did with postcss-markdown and postcss-css-in-js.

Please consider helping out if you've time as I believe this issue is going to need some poking around.

---

As an aside, @ota-meshi has been working on [experimental support for stylus](https://www.npmjs.com/package/stylelint-plugin-stylus). However, I think we need to resolve how the [`customSyntax` option](https://stylelint.io/user-guide/usage/options#customsyntax) relates to embedded styles. The waters are muddy at the moment as we have syntaxes within syntaxes, e.g. stylus with HTML.

**Comment by Mister-Hope** [comment]
I am using the workspace settings in vscode to disable the checking by setting `stylelint.validate` without `'vue'` as a workaround,

But I am afraid I do not have much time now. As a grade 4 undergraduate student I am busy with my  thesis now.

**Comment by Mister-Hope** [comment]
Any update about this?

**Comment by jeddy3** [comment]
The `postcss-html` custom syntax will now silently ignore style blocks with a `lang` it doesn't have an inner custom syntax for.

For example:

```
npm i postcss-html stylelint
```

```html
<style lang="stylus">
  a
    color: #ff
</style>
```

```json
{
  "customSyntax": "postcss-html",
  "rules": {
    "color-no-invalid-hex": true
  }
}
```

```console
jeddy3@MacBook stylelint-test % npx stylelint test.html
jeddy3@MacBook stylelint-test % 
```

If we install `postcss-styl`:

```
npm i postcss-styl
```

We get:

```console
jeddy3@MacBook stylelint-test % npx stylelint test.html

test.html
 3:12  ✖  Unexpected invalid hex color "#ff"  color-no-invalid-hex
```
